Casks/n/nx-studio.rb

```diff
@@ -11,6 +11,23 @@
   livecheck do
     url "https://downloadcenter.nikonimglib.com/en/products/564/NX_Studio.html"
     regex(/Ver\.\s+(\d+(?:\.\d+)+)/i)
+    strategy :page_match do |page, regex|
+      version = page.scan(regex).filter_map { |match| match.first.tr(" ", "") }&.first
+
+      html_id = page.scan(%r{/download/sw/(\d+)\.html}i)&.first
+      next if html_id.blank?
+
+      version_page = Homebrew::Livecheck::Strategy.page_content("https://downloadcenter.nikonimglib.com/en/download/sw/#{html_id[0]}.html")
+      download_query = version_page[:content]&.match(/id="controllerMac".*redirect.do\?(.*?)[" <]/im)
+      next if download_query.blank?
+
+      redirect_url = "https://crossgate.nikonimglib.com/dsd_redirect/redirect.do?#{download_query[1].gsub("&amp;", "&")}"
```

Comment:
```suggestion
      # Match the redirecting URL for the macOS file
      download_redirect_url = version_page[:content]&.[](
        /controllerMac.*?href=["']?([^"' >]+?redirect\.do\?[^"' >]+?)["' >]/im, 1
      )&.gsub("&amp;", "&")
      next if download_redirect_url.blank?

```

The redirect URL is on a different domain, so presumably it will continue to be an absolute URL. As long as that remains true, there's something to be said for matching the full URL from the `href` attribute and using that instead of interpolating only one part into a URL string, which would break if any part of the URL differs from what we have in the `strategy` block. Using the full URL can still fail in other ways but it's potentially less fragile in this respect.